project/assets/scripts/controller/new-evaluation/new-evaluation.js

```diff
@@ -1,68 +1,78 @@
+            document.getElementById('evaluations').classList.remove('navigation-bar-active');
+            document.getElementById('new-evaluation').classList.add('navigation-bar-active');
+            setNavEvent('logout', addEvent.bind(null, 'login', redirectToFunction));
 
-        const submitBtn = document.getElementById('submit-button');
-        submitBtn.addEventListener('click', getFormData);
+            const submitBtn = document.getElementById('submit-button');
+            submitBtn.addEventListener('click', getFormData);
+        };
+        this.destroy = function (redirectToFunction) {
+            removeEvents('evaluations', addEvent.bind(null, 'evaluations', redirectToFunction));
+            removeEvents('logout', addEvent.bind(null, 'login', redirectToFunction));
+            removeEvents('submit-button', getFormData);
+        };
     };
-    this.destroy = function (redirectToFunction) {
-        removeEvents('evaluations', addEvent.bind(null, 'evaluations', redirectToFunction));
-        removeEvents('logout', addEvent.bind(null, 'login', redirectToFunction));
-        removeEvents('submit-button', getFormData);
-    };
-};
 
-function getSelected(selectId) {
-    const e = document.getElementById(`${selectId}`);
-    return e.options[e.selectedIndex].value;
-}
+    function getSelected(selectId) {
+        const e = document.getElementById(`${selectId}`);
+        return e.options[e.selectedIndex].value;
+    }
 
-function getTextarea() {
-    const textareaArray = getTextareaArray().items.map((el) => {
-        const textareaKey = constructId(el.title);
-        const textareaObj = {};
-        textareaObj[textareaKey] = document.getElementById(constructId(el.title)).value;
-        return textareaObj;
-    });
-    return textareaArray;
-}
+    function getTextarea() {
+        const textareaArray = feedbackApp.data.newEvaluation.textareaItems().items.map((el) => {
+            const textareaKey = constructId(el.title);
+            const textareaObj = {};
+            textareaObj[textareaKey] = document.getElementById(constructId(el.title)).value;
+            return textareaObj;
+        });
+        return textareaArray;
+    }
 
-function getFieldset() {
-    const fieldsetArr = getTechnicalAreaArray().map((el) => {
-        const obj = {};
-        const objKey = constructId(el.legend);
-        obj[objKey] = {};
-        const dropDownObj = {};
-        el.dropDownOptions.forEach((op) => {
-            op.forEach((i) => {
-                const dropDownKey = constructIdFieldset(el.legend, i.label);
-                dropDownObj[dropDownKey] = getSelected(constructIdFieldset(el.legend, i.label));
-                obj[objKey] = dropDownObj;
+    function getFieldset(data) {
+        const fieldsetArr = data.map((el) => {
+            const obj = {};
+            const objKey = constructId(el.legend);
+            obj[objKey] = {};
+            const dropDownObj = {};
+            el.dropDownOptions.forEach((op) => {
```

Comment:
This whole thing smells a bit like a `reduce` to me. :)

Reply:
Challenge accepted!